pkg/autodiscovery/listeners/kube_services.go

```diff
@@ -174,6 +213,7 @@ func (l *KubeServiceListener) createService(ksvc *v1.Service, firstRun bool) {
 
 	l.m.Lock()
 	l.services[ksvc.UID] = svc
+	l.endpointsAnnotated[fmt.Sprintf("%s/%s", ksvc.Namespace, ksvc.Name)] = isEndpointAnnotated(ksvc)
```

Comment:
This won't be called if a service has `ad.datadoghq.com/endpoints` annotations but no `ad.datadoghq.com/service` annotations. We might want to move it up higher in the flow